### PR TITLE
Move default get-completions to sh.janet and add tests

### LINKED ISSUE #184: Tests for completion functions
As the completions have got more advanced, we should be testing them with a test case. It may be easiest to shift the default completions to sh.janet to make them easy to access, then it will be easy to just make a normal test case file for them.

### PR DESCRIPTION
**Pull request by ALSchwalm**
Fixes #184

Definitely possible there are some more test cases I'm missing. If so please point them out. I also had to call out to chmod to set the executable bit (kind of wondering if there should be a os/chmod even though that's probably a portability nightmare). Hopefully that is available in the runner.

Edit: Oh, it looks like the interface to os/execute has changed pretty recently. Not sure what the right play is here. Can the runner janet version be updated?

### CODE REVIEW

File: test/cases/0005-completions [hunk 1]
```diff
@@ -0,0 +1,63 @@
+#! /usr/bin/env janetsh
+(import sh)
+
+# FIXME use an assert macro that
+# prints what we wanted and what we got.
+
+(var env (fiber/getenv (fiber/current)))
+
+(defn unique [arr]
```
Comment (andrewchambers) [inline test/cases/0005-completions]: Std library has 'distinct'

**Comment by andrewchambers** [comment]
Sorry, I somehow edited your comment by mistake

**Comment by andrewchambers** [comment]
Fixes #184 

> I also had to call out to `chmod` to set the executable bit (kind of wondering if there should be a `os/chmod` even though that's probably a portability nightmare). Hopefully that is available in the runner.

It should be fine, chmod will be everywhere. I agree it could be in the stdlib, we could also just put a c version in shlib, though just for a test might not be needed.

> Can the runner janet version be updated?

Yup, the janet version used by the runners is defined in support/ci.sh and is easy to update.

**Comment by ALSchwalm** [comment]
Sorry for all the force pushes. It looks like recent commits on Janet fail to build on BSD (I'm guessing since this: https://github.com/janet-lang/janet/commit/c43aaf898608840051378e77da8b5bddeaa60bee). But I also need the more recent commit for passing custom environments to `all-bindings`, so there is no version of Janet that will make this work atm.

**Comment by andrewchambers** [comment]
@ALSchwalm  I just told @bakpakin  on gitter.

**Comment by andrewchambers** [comment]
@ALSchwalm  new janet commit to try.

**Comment by ALSchwalm** [comment]
Yep, that fixed it. Thanks!

**Comment by andrewchambers** [comment]
Thanks for doing this tidying work, I really appreciate it.

**Pull request merged**